### PR TITLE
Bagger in production environment

### PR DESCRIPTION
**Pull request by tomcrane**
### What is this PR trying to achieve?

The METS bagger running on AWS and listening for jobs
https://github.com/wellcometrust/platform/issues/2635

### Who is this change for?

Platform team, for anyone to be able to run sets of METS through the archive bagging service for testing

### CODE REVIEW

File: archive/terraform/main.tf [hunk 1]
```diff
@@ -55,3 +55,53 @@ module "registrar" {
...
+    DROP_BUCKET_NAME_ERRORS     = "${var.bagger_drop_bucket_name_errors}"
+    CURRENT_PRESERVATION_BUCKET = "${var.bagger_current_preservation_bucket}"
+    DLCS_SOURCE_BUCKET          = "${var.bagger_dlcs_source_bucket}"
+    BAGGING_QUEUE               = "${module.bagger_queue.name}"
+
+    # aws creds that can see Preservica as well as Platform estate
+    AWS_ACCESS_KEY_ID     = "${var.bagger_aws_access_key_id}"
+    AWS_SECRET_ACCESS_KEY = "${var.bagger_aws_secret_access_key}"
+    AWS_DEFAULT_REGION    = "${var.aws_region}"
+
```
Comment (mklander) [inline archive/terraform/main.tf]: Can this be done by a role?

File: archive/bagger/.gitignore [hunk 2]
```diff
@@ -106,6 +106,10 @@ venv.bak/
 *.bat
 bag.sh
 bnums.sh
+local_run.sh
+queue_for_bagging.sh
+local_run_queue_listener.sh
+error_report.sh
 .vscode
```
[the remark this answers is not in the document]
Comment (tomcrane) [reply archive/bagger/.gitignore]: Yes, they are for running various parts of this code locally, they provide the python code with its environment vars.

File: archive/terraform/main.tf (hunk 1 above)
Comment (tomcrane) [reply archive/terraform/main.tf]: Yes, we should adapt that bit to provide those vars following the convention of the rest of the platform.

File: archive/terraform/main.tf [hunk 3]
```diff
@@ -55,3 +55,53 @@ module "registrar" {
...
+    DLCS_API_KEY     = "${var.bagger_dlcs_api_key}"
+    DLCS_API_SECRET  = "${var.bagger_dlcs_api_secret}"
+    DLCS_CUSTOMER_ID = "${var.bagger_dlcs_customer_id}"
+    DLCS_SPACE       = "${var.bagger_dlcs_space}"
+
+    # DDS credentials
+    DDS_API_KEY      = "${var.bagger_dds_api_key}"
+    DDS_API_SECRET   = "${var.bagger_dds_api_secret}"
+    DDS_ASSET_PREFIX = "${var.bagger_dds_asset_prefix}"
+  }
```
Comment (mklander) [inline archive/terraform/main.tf]: Similar here, we should check with @kenoir whether it would be better to manage these values in another way.

**Comment by kenoir** [comment]
@tomcrane can we get this mergeable and add anything else in a different PR?

**Comment by tomcrane** [comment]
It should be mergeable now, still need at address @mklander's concern above, will look at that with @fractos

File: archive/terraform/main.tf (hunk 3 above)
Comment (fractos) [reply archive/terraform/main.tf]: @mklander These keys are for an external service (the DLCS), so it won't be possible to carry them by IAM role.

**Pull request merged**